### PR TITLE
Update to_sparse docs regarding the layout and blocksize kw arguments.

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89912

cc @nikitaved @cpuhrsch @amjames @bhosmer @svekars @carljparker

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -5378,6 +5378,47 @@ def callable(a, b) -> number
...
+
+    layout (:class:`torch.layout`, optional): The desired sparse
+      layout. One of ``torch.sparse_coo``, ``torch.sparse_csr``,
+      ``torch.sparse_csc``, ``torch.sparse_bsr``, or
+      ``torch.sparse_bsc``. Default: if ``None``,
+      ``torch.sparse_coo``.
+
+    blocksize (list, tuple, :class:`torch.Size`, optional): Block size
+      of the resulting BSR or BSC tensor. For other layouts,
+      specifying the block size that is not ``None`` will result a
```
Comment (cpuhrsch) [inline torch/_tensor_docs.py:5402]: nit: "will result *in* a RuntimeError"

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -5378,6 +5378,47 @@ def callable(a, b) -> number
...
+      RuntimeError exception.  A block size must be a 2-sequence and
```
Comment (cpuhrsch) [inline torch/_tensor_docs.py:5403]: Can't we just say "tuple of length two" and "evenly divide the sparse dimensions"?

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -5378,6 +5378,47 @@ def callable(a, b) -> number
...
+      compatible with the shape of :attr:`self`: ``self.shape[B] %
+      blocksize[0] == self.shape[B + 1] % blocksize[1] == 0`` where
+      ``B`` is the number of batch dimensions if :attr:`self` is a
+      sparse CSR/CSC/BSR/BSC tensor, otherwise, ``B = 0``.
+
+Example::
```
Comment (cpuhrsch) [inline torch/_tensor_docs.py:5409]: I'd also add an example of where blocksize fails

File: torch/_tensor_docs.py (hunk 2 above)
Comment (pearu) [reply torch/_tensor_docs.py:5403]: How about this:
```
A block size must be a tuple of length two such that its items are evenly divisible with the two sparse dimensions.
```
?

**Review by cpuhrsch — APPROVED** [review]

File: torch/_tensor_docs.py (hunk 2 above)
Comment (pearu) [reply torch/_tensor_docs.py]: Fixed.

File: torch/_tensor_docs.py (hunk 1 above)
Comment (pearu) [reply torch/_tensor_docs.py]: Fixed.

**Pull request closed without merging**